core/trino-spi/src/main/java/io/trino/spi/connector/Connector.java

```diff
@@ -147,6 +147,11 @@ default Set<TableProcedureMetadata> getTableProcedures()
         return emptySet();
     }
 
+    default long getExtraMemoryRequirement()
```

Comment:
Why not make it `Supplier<Long>` and let engine poll that periodically?

This would allow for example Iceberg connector to report memory used for storing deletion files data, something @sopel39 was concerned about.

Reply:
or maybe a pushed-based approach rather than polling; this is secondary.
main point is to allow allocations that change over time.

Reply:
it's too simple for use-cases that we can name today

Reply:
> it's too simple for use-cases that we can name today

I don't think so. Consider some internal caching structures/buffers. Then it's simple enough

Reply:
i was thinking about iceberg deletion files caches.

Reply:
> i was thinking about iceberg deletion files caches.

Then you probably needs some revocable `MemoryAllocator` passed here. But `getExtraMemoryRequirement` don't exclude `MemoryAllocator` approach later on

Reply:
revocable is nice, but not required for none of the use-cases existing today that i know about.

Reply:
> revocable is nice, but not required for none of the use-cases existing today that i know about.

Then I think `getExtraMemoryRequirement` is good enough even for delete file cache (just fixed pool size, not too big because it's more per query kind of cache atm)